packages/core/src/fhir-to-cda/cda-templates/components/__tests__/medications.test.ts

```diff
@@ -0,0 +1,190 @@
+          },
+        ],
+      },
+    ],
+  };
+
+  medication = {
+    resourceType: "Medication",
+    id: medId,
+    code: {
+      text: "CARVEDILOL",
+      coding: [
+        {
+          code: "315577",
+          display: "carvedilol 6.25 MG",
+          system: "2.16.840.1.113883.6.88",
+        },
+        {
+          code: "51407-040",
+          display: "Carvedilol",
+          system: "2.16.840.1.113883.6.69",
+        },
+      ],
+    },
+  };
+
+  bundle = createEmptyBundle();
+});
+
+describe("buildMedications", () => {
+  it("throws an error if the Bundle is missing a Medication", () => {
+    bundle.entry?.push({ resource: medicationStatement });
+    expect(() => buildMedications(bundle)).toThrow();
+  });
+
+  it("throws an error if MedicationStatement incorrectly references a Medication", () => {
+    bundle.entry?.push({ resource: medicationStatement });
+    const medicationWithAlteredId = {
+      ...medication,
+      id: "SomethingElse",
+    };
+    bundle.entry?.push({ resource: medicationWithAlteredId });
+    expect(() => buildMedications(bundle)).toThrow();
+  });
+
+  it("correctly maps a single MedicationStatement with a related Medication", () => {
+    bundle.entry?.push({ resource: medicationStatement });
+    bundle.entry?.push({ resource: medication });
+
+    expectedXml = `<component><section><templateId root="2.16.840.1.113883.10.20.22.2.1.1"></templateId><code code="10160-0" codeSystem="2.16.840.1.113883.6.1" codeSystemName="LOINC" displayName="History of Medication use Narrative"></code><title>MEDICATIONS</title><text><table ID="medications"><thead><tr><th>Medication</th><th>Code</th><th>Dosage</th><th>Frequency</th><th>Start Date</th><th>End Date</th><th>Reason</th></tr></thead><tbody><tr ID="medications1"><td>CARVEDILOL</td><td>315577 - 2.16.840.1.113883.6.88</td><td>6.25 mg (milligram)</td><td>1 pill/day</td><td>${pastDateHumanReadable}</td><td>${NOT_SPECIFIED}</td><td>heart</td></tr></tbody></table></text><entry><substanceAdministration classCode="SBADM" moodCode="INT"><templateId root="2.16.840.1.113883.10.20.22.4.16" extension="2014-06-09"></templateId><id root="placeholder-ORG-OID" extension="${medStmntId}"></id><statusCode code="active"></statusCode><effectiveTime xsi:type="IVL_TS"><low value="${pastDateXml}"></low></effectiveTime><consumable typeCode="CSM"><manufacturedProduct><templateId root="2.16.840.1.113883.10.20.22.4.23" extension="2014-06-09"></templateId><manufacturedMaterial><code code="315577" codeSystem="2.16.840.1.113883.6.88" displayName="carvedilol 6.25 MG"><originalText><reference value="medications1"></reference></originalText><translation code="315577" codeSystem="2.16.840.1.113883.6.88" displayName="carvedilol 6.25 MG"></translation><translation code="51407-040" codeSystem="2.16.840.1.113883.6.69" displayName="Carvedilol"></translation></code></manufacturedMaterial></manufacturedProduct></consumable></substanceAdministration></entry></section></component>`;
```

Comment:
ditto, keeping large payloads separated from the code.

In this case, we could:
1. store this as a `.xml` - to help w/ editing/reviewing;
2. load in memory
3. run it as JS code with `eval()`

Something like:

```js
const templ = "<xml>${content}</xml>";
const content = "xuxu"
const res = eval("`"+templ+"`");
console.log(res)
console.log(Array.from(res).join(","))
console.log(`<xml>${content}</xml>`);
```

JSFiddle: https://jsfiddle.net/6bztcu3y/4/

Reply:
Sounds good! Will use this method